### DISCUSSION TITLE
Glob astro files does not include frontmatter

### DISCUSSION THREAD

**Discussion by ivoba** [withastro/roadmap#402] [Proposal]
Is there areason why frontmatter is not included in glob results when globbing over *.astro files?

    const pages = await Astro.glob("./pages/*.astro");

https://docs.astro.build/en/reference/api-reference/#astro-files

```
export interface AstroInstance {
  /* The file path of this file */
  file: string;
  /* The URL for this file (if it is in the pages directory) */
  url: string | undefined;
  default: AstroComponent;
}
```

I would have thought it would return the same as for *md or *mdx files.

**Comment by delucis** [comment]
The "frontmatter" of Astro files is quite different from Markdown & MDX because it's JavaScript rather than YAML. What would you expect to get when globbing a `.astro` file?

**Reply by ivoba** [reply]
Yes you are right. Astro file frontmatter is javascript.

I thought i could glob over the astro files and build a navigation just as with *.md files.
But then infos are missing like title which would be described inside the globbed .astro file.

**Reply by jasikpark** [reply]
This would be an interesting usecase for allowing exporting data from the frontmatter, so that 

```astro
---
export const title = "Hello World"
export const description = "A blog post xyz"
---

this is my astro component content
```

could work. `import { title, description } from "HelloWorldBlogPost.astro";`

**Reply by jasikpark** [reply]
true astro mdx xD

**Reply by ivoba** [reply]
@jasikpark awesome! this works.

in /pages/subpages/post.astro
```
---
export const title = "Hello World"
export const description = "A blog post xyz"
---
```

in /pages/list.astro

```
---
const astroPages = await Astro.glob("./subpages/*.astro");
---

<main>
        {
          astroPages.map(({ url, title, description }) => (
            <Card
              page={url || ""}
              title={title}
              body={shortDesc}
            />
          ))
        }
</main>
```

The reason for this is a very nested pages structure.
I got bitten here by the limitations of glob only taking literals and routing did not picking up nested structure correctly.

VsCode complains though:
> Property 'title' does not exist on type 'AstroInstance'.

**Reply by Princesseuh** [reply]
We purposely currently do not document this / support it officially because there's some limitations that are hard to explain / document / support in regard to how the exports are hoisted. We recommend not relying on it too much 😅

**Comment by florian-lefebvre** [comment]
`Astro.glob` is deprecated in v5: https://docs.astro.build/en/guides/upgrade-to/v5/#deprecated-astroglob
